slsim/lens.py

```diff
@@ -1359,6 +1367,26 @@ def generate_id(self, ra=None, dec=None):
 
         return f"{lens_type}-LENS_{ra:.4f}_{dec:.4f}"
 
+    def add_subhalos(
+        self, source_index=0
+    ):  # , return_stats=False, suppress_output=True
+        """Get subhalo models from pyHalo and add them to the lens model.
+
+        :param los_norm: rescales the amplitude of the line-of-sight
+            halo mass function
+        :param source_index: index of source, default =0, i.e. the first
+            source
+        """
+        z_lens = self.deflector_redshift
+        z_source = self.max_redshift_source_class.redshift
+        einstein_radius = self._get_effective_einstein_radius(source_index)
+        cone_opening_angle = 4 * einstein_radius
+        realization = CDM(
+            z_lens, z_source, cone_opening_angle_arcsec=cone_opening_angle
```

Comment:
can you pass a dictionary into the function for other optional choices for the user to be passed into phHalo?